scala-server-quality-assurance.md

```diff
@@ -0,0 +1,174 @@
+# Scala Server project Quality Assurance
+
+The goal of this repository is to define general guidance for server-side projects written in Scala at Karumi. This document complements the [Project Quality Assurance](./README.md) with all the specifics we follow for server-side projects written in Scala.
+
+## Feature kickstart
+
+We all love to start new features. These are a few questions you should ask yourself first to prevent potential delays.
+
+* Are the feature, functional and non-functional, requirements ready?
+  - [ ] Does the feature description include all the information needed to start designing the database usage?
+  - [ ] Is it designed keeping in mind the format and structure of the data?
+  - [ ] Can we reuse part of the already implemented platform?
+  - [ ] Do the user stories describe the usage of the feature from the back office or the admin panel?
+  - [ ] Does it include a description of different scenarios related to corner or edge cases?
+  - [ ] Does it contain any non-functional requirement related to performance or any resource usage?
+  - [ ] Does it include examples needed to validate the feature form the acceptance criteria viewpoint?
+  - [ ] In case of implementing any feature with any user interface component like emails or the back office user interaction, does the user stories provide links to the sketch design or assets?
+
+* Localization
+  - [ ] Are the translations for this feature defined? If not, require them; if they are, consider if there is enough room to display them, especially the dynamic ones (the ones displaying usernames, amounts of money, etc.)
+
+* Are the error scenarios well defined?
+  - [ ] The client is just sending the mandatory information, what's the expected behavior?
+  - [ ] The client is not sending the requested data correctly, what would it happen?
+  - [ ] Third-party services are down, how should the server behave?
+  - [ ] Are all the business errors adequately handled? If not, should they be?
+
+## Pull requests
+
+* Guide the reviewers in how to test the new feature, the endpoints or sites that are affected, how to access them. If there is any context requirement (a user logged in, malfunctioning third-party service, etc.) specify how did you test it.
+
+  * All pull requests must have tests of the endpoints or screens showing UI changes if any.
+
+  * If needed, specify if there are changes in any DB schema and how the evolution is performed.
+
+  * Review if any particular index needs to be configured to improve the service performance. If this is the case, specify it and explain the reason why this index is necessary.
+
+  * We should indicate if any infrastructure change is required.
```

Comment:
Aren't we using terraform for infrastructure configuration? I'd add a reference in the tooling section if that's the case